patches/sitecustomize.py

```diff
@@ -1,22 +1,56 @@
 # TODO(rosbo): Remove this once we fix the issue with fastai importing older libcudnn if imported prior to tensorflow
 import tensorflow
 import os
+from google.auth import credentials
+from google.cloud import bigquery
+from google.cloud.bigquery._http import Connection
 
-# Monkey patches BigQuery client creation to use proxy.
-kaggle_proxy_token = os.getenv("KAGGLE_DATA_PROXY_TOKEN")
-if kaggle_proxy_token:
-    from google.auth import credentials
-    from google.cloud import bigquery
-    from google.cloud.bigquery._http import Connection
+class KaggleKernelCredentials(credentials.Credentials):
+    def refresh(self, request):
+        print("Calling Kaggle.UserSecrets to refresh token.")
+        # Set self.token and self.expiry here.
+        raise NotImplementedError("Private BigQuery integration is not yet implemented.")
 
-    Connection.API_BASE_URL = os.getenv("KAGGLE_DATA_PROXY_URL")
-    Connection._EXTRA_HEADERS["X-KAGGLE-PROXY-DATA"] = kaggle_proxy_token
+kaggle_proxy_token = os.getenv("KAGGLE_DATA_PROXY_TOKEN")
+CONNECTION_BASE_URL = Connection.API_BASE_URL
+def monkeypatch_bq(bq_client, *args, **kwargs):
+    data_proxy_project = os.getenv("KAGGLE_DATA_PROXY_PROJECT")
```

Comment:
Consider passing `data_proxy_project` and `bq_user_jwt` in the kwargs of this method. At line 55, just set the default values for these parameters with `os.getenv(...)`. 

This way, in the tests, you can override them without having to mess with the environment variable.

Reply:
Do you mean, then I would pass those in the bigquery client creation in the test? Ie. client = bigquery.Client(project='KAGGLE', bq_user_jwt=foo)? The only reason I don't like that is that it modifies the interface of the bigquery client library just for testing purposes. Let me see if I can come up with something else that works otherwiwse I'll do this :) 
By the way, would there be an issue with change the env var in the tests (and then clean them up to what they were)? In particular, if tests run in parallel I could see that being an issue. I almost want to have a integration style test for this where it does a separate docker run for each test since it'll be a clean environment each time and any modified global state of code from sitecustomize.py from one test won't affect the other.

Reply:
I am mostly concerned about the fact that we run our tests in parallel.

Reply:
Are you planning to address this in a separate PR or this PR? (either way is fine with me).

Reply:
Just did it in the latest commit :) Learned something new today: EnvironmentVarGuard is pretty handy!

Reply:
fyi tests don't run in parallel (unless something changed).  Still 👍 on the change though.